### PR TITLE
adding query for highest count of other ids

### PR DESCRIPTION
**Pull request by joker2411**
## Description of the change

> Ticket [Link](https://linear.app/rudderstack/issue/PRML-955/1-add-drews-feedback-to-id-stitcher-analysis-tool).

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py [hunk 1]
```diff
@@ -391,11 +393,32 @@ def run(self):
         for row in result.to_dict(orient="records"):
             print(f"\t\t{row['OTHER_ID_TYPE']}: {row['AVG_COUNT']}")
 
+        # Main-ids with the highest count of other ids, for each id type
+        query = f"""
+        SELECT other_id_type, {main_id_key}, max_count
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py:398]: Why not just have max(count) in prev query itself?

Comment (dpatchigolla) [reply src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py:398]: We don't need to get the main_id with highest count here.

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py [hunk 2]
```diff
@@ -215,12 +215,12 @@ def check_missing_connections(self, node_types: List[str]) -> List[str]:
         # missing indirect connections too.
         if warn:
             print(
-                "Following id types are defined in id stitcher graph to come from same table, but we do not have any direct edges between them in the data (indirect edges can exist)."
+                "Following id types are defined in id stitcher graph before id-stitching to come from same table, but we do not have any direct edges between them in the data (indirect edges can exist)."
```
Comment (dpatchigolla) [inline src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py:218]: Lets reword to

`Following id types are defined in id stitcher model inputs to come from same table, but we never see them together before id-stitching (they may be linked later by the stitching through other ids though).`

**Review by dpatchigolla — CHANGES_REQUESTED** [review]

File: src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py (hunk 1 above)
Comment (joker2411) [reply src/predictions/profiles_mlcorelib/py_native/id_stitcher/table_report.py:398]: ok.

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**